InscryptionAPI/Card/AbilityManager.cs

```diff
@@ -249,4 +251,54 @@ public static void FixRulebook(AbilityMetaCategory metaCategory, RuleBookInfo __
             }
         }
     }
+
+    [HarmonyPatch(typeof(GlobalTriggerHandler), nameof(GlobalTriggerHandler.TriggerCardsOnBoard))]
+    [HarmonyPostfix]
+    public static IEnumerator WaterborneFix(IEnumerator result, Trigger trigger, bool triggerFacedown, params object[] otherArgs)
+    {
+        yield return result;
+        if (!triggerFacedown)
+        {
+            bool RespondsToTrigger(CardTriggerHandler r, Trigger trigger, params object[] otherArgs)
+            {
+                foreach (TriggerReceiver receiver in r.GetAllReceivers())
+                {
+                    if (GlobalTriggerHandler.ReceiverRespondsToTrigger(trigger, receiver, otherArgs) && ((receiver is ActivateWhenFacedown && (receiver as ActivateWhenFacedown).ShouldTriggerWhenFaceDown(trigger, otherArgs)) || 
+                        (receiver is ExtendedAbilityBehaviour && (receiver as ExtendedAbilityBehaviour).TriggerWhenFacedown)))
+                    {
+                        return true;
+                    }
+                }
+                return false;
+            }
+            IEnumerator OnTrigger(CardTriggerHandler r, Trigger trigger, params object[] otherArgs)
+            {
+                foreach (TriggerReceiver receiver in r.GetAllReceivers())
+                {
+                    if (GlobalTriggerHandler.ReceiverRespondsToTrigger(trigger, receiver, otherArgs) && ((receiver is ActivateWhenFacedown && (receiver as ActivateWhenFacedown).ShouldTriggerWhenFaceDown(trigger, otherArgs)) ||
+                            (receiver is ExtendedAbilityBehaviour && (receiver as ExtendedAbilityBehaviour).TriggerWhenFacedown)))
+                    {
+                        yield return Singleton<GlobalTriggerHandler>.Instance.TriggerSequence(trigger, receiver, otherArgs);
+                    }
+                }
+                yield break;
+            }
+            List<PlayableCard> list = new(Singleton<BoardManager>.Instance.CardsOnBoard);
+            foreach (PlayableCard playableCard in list)
+            {
+                if (playableCard != null && playableCard.FaceDown && RespondsToTrigger(playableCard.TriggerHandler, trigger, otherArgs))
+                {
+                    yield return OnTrigger(playableCard.TriggerHandler, trigger, otherArgs);
+                }
+            }
+        }
+        yield break;
+    }
+}
+
+public interface ActivateWhenFacedown
```

Comment:
One last nitpick - this should be in its own file.

Reply:
done